src/nimlsp.nim

```diff
@@ -156,6 +174,17 @@ while true:
           gotShutdown = true
         of "initialize":
           debugEcho "Got initialize request, answering"
+          
+          let params = message["params"].unsafeGet
+          var projectRoot,projectFile:string
+          if params.hasKey("workspaceFolders"):
```

Comment:
Maybe whenValid is more apt? Why not?

Reply:
I thought may like `message.textDocumentRequest(CompletionParams, compRequest):` that accept  a InitializeParams